test/commands/init.test.js

```diff
@@ -0,0 +1,270 @@
+            execSync("npm install");
+        }
+    }
+
+    function makeOclifConfig(config) {
+        config.loadPlugins = async function(path, type, plugins) {
+            plugins.forEach(pluginOpts => this.plugins.push({ options: pluginOpts }));
+        };
+        config.plugins = config.plugins || [];
+        config.commands = config.commands || [];
+        return config;
+    }
+
+    it("uses the devDependency of the project if run by aio", async function() {
+        const config = makeOclifConfig({
+            pjson: {
+                name: "@adobe/aio-cli" // not standalone
+            },
+            plugins: [{
+                options: {
+                    type: "user",
+                    root: GLOBAL_INSTALLED_PLUGIN_ROOT,
+                    name: "@adobe/aio-cli-plugin-asset-compute",
+                }
+            }, {
+                options: {
+                    type: 'core',
+                    root: '/usr/local/lib/node_modules/@adobe/aio-cli',
+                    name: '@adobe/aio-cli-plugin-events'
+                }
+            }]
+        });
+
+        cdProjectAndInstall(TEST_PROJECT_WITH_DEVDEPENDENCY);
+        await init({ config });
+
+        // verify previous plugin was removed
+        assert.ok(config.plugins.every(plugin => plugin.options.root !== GLOBAL_INSTALLED_PLUGIN_ROOT), "global installed plugin still present");
+
+        // verify local devDependency was added as plugin
+        assert.ok(config.plugins.some(
+            plugin => plugin.options.name === "@adobe/aio-cli-plugin-asset-compute"
+                      && plugin.options.root === path.resolve("node_modules/@adobe/aio-cli-plugin-asset-compute")
+        ), "local devDependency plugin is missing");
+
+    }).timeout(30000); // because of the npm install
+
+    it("adds itself even if it can't find itself in the plugin list", async function() {
+        const config = makeOclifConfig({
+            pjson: {
+                name: "@adobe/aio-cli" // not standalone
+            },
+            plugins: [{
+                options: {
+                    type: 'core',
+                    root: '/usr/local/lib/node_modules/@adobe/aio-cli',
+                    name: '@adobe/aio-cli-plugin-events'
+                }
+            }]
+        });
+
+        cdProjectAndInstall(TEST_PROJECT_WITH_DEVDEPENDENCY);
+        await init({ config });
+
+        // verify previous plugin was removed
+        assert.ok(config.plugins.every(plugin => plugin.options.root !== GLOBAL_INSTALLED_PLUGIN_ROOT), "global installed plugin still present");
+
+        // verify local devDependency was added as plugin
+        assert.ok(config.plugins.some(
+            plugin => plugin.options.name === "@adobe/aio-cli-plugin-asset-compute"
+                      && plugin.options.root === path.resolve("node_modules/@adobe/aio-cli-plugin-asset-compute")
+        ), "local devDependency plugin is missing");
+    });
```

Comment:
this one does not do an `npm install` so it does not need the extra time for the tests?

Reply:
Yeah, the first test will have done the `npm install` already. The `node_modules` isn't deleted between them and the npm install is only done if node_modules doesn't exist yet: https://github.com/adobe/aio-cli-plugin-asset-compute/pull/34/files/07b196baaa6d0b3ab6e83c6667292f5ad6b70ba6#diff-e3f3918f471c6e18dd2e47a71a1d3971R30-R31

Hence the following tests are faster. And this is the purpose, if we'd reset it (delete node_modules) for every test it would all be much slower.